janitor/utils.py

```diff
@@ -2373,3 +2377,805 @@ def _sub_process_text_result_MultiIndex(index: pd.MultiIndex, result, df):
+    # is equal to the length of `right_keys`
+    # that means the respective position in `left_c`
+    # has no values from `right_c` that are less than
+    # or equal, and should therefore be discarded
+    len_right = right_keys.size
+    rows_equal = search_indices == len_right
+    if rows_equal.any():
+        left_c = left_c[~rows_equal]
+        search_indices = search_indices[~rows_equal]
+
+    # if there are any rows where left === right
+    # then the search index position should be moved
+    # forward by 1 ... this way we keep only values
+    # that are strictly less than -
+    # when the indices are created by _le_create_ranges,
+    # the indices generate will be strictly positions
+    # that are less than
+    # exact comparision is used here, instead of `isin`,
+    # as isin can give silent errors when comparing ints
+    # to floats
+    if strict:
+        rows_equal = right_keys.take(search_indices).array
+        rows_equal = left_c == rows_equal
+        if rows_equal.any():
+            search_indices = np.where(
+                rows_equal, search_indices + 1, search_indices
+            )
+            # it is possible for index positions in search_indices
+            # become same size as `right_keys`, due to the shift
+            # from the code above
+            rows_equal = search_indices == len_right
+            if rows_equal.any():
+                left_c = left_c[~rows_equal]
+                search_indices = search_indices[~rows_equal]
+    # if search_indices is empty, exit
+    if search_indices.size == 0:
+        return None, None
+
+    # for each index in `search_indices`,
+    # generate all indices for `right_keys`,
+    # where the values in `right_keys` are greater than
+    # or equal to `left_c`, depending on the value of `strict`
+    positions = _le_create_ranges(search_indices, len_right)
+
+    # get the index locations for the unique values in `right_c`
+    right_indices = right_group.indices
+    # map the indices to `right_keys`
+    # and subsequently concatenate to get
+    # the complete matched values from `right_c`
+    right_positions = right_keys.take(positions)
+    right_c = right_positions.map(right_indices)
+    right_c = np.concatenate(right_c)
+
+    # Index.repeat twice
+    # first to align left_c to right_c
+    # second time, it is to align to the total combination
+    # of indices from right_c
+    left_c = left_c.index.repeat(len_right - search_indices)
+    search_indices = right_positions.map(right_group.size())
+    left_c = left_c.repeat(search_indices)
+    return left_c, right_c
+
+
+def _greater_than_indices(
+    left_c: pd.Series, right_c: pd.Series, strict: bool = False
+):
+    """
+    This uses binary search to get indices where left_c is greater than
+    or equal to right_c. If strict is True, then only indices where `left_c`
+    is greater than (but not equal to) `right_c` are returned.
+
+    Returns a tuple of (left_c, right_c)
+    """
```

Comment:
Minor grammatical nitpick.

```suggestion
    """
    Use binary search to get indices where left_c is greater than
    or equal to right_c. If strict is True, then only indices where `left_c`
    is greater than (but not equal to) `right_c` are returned.

    Returns a tuple of (left_c, right_c)
    """
```